### PR TITLE
Added `TypeList`.

### PR DESCRIPTION
**Pull request by dkorolev**
Baby steps ... :-)

Thanks,
Dima

### CODE REVIEW

File: Bricks/template/typelist.h [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+static_assert(std::is_same<TypeListImpl<int>, Flatten<TypeListImpl<TypeListImpl<TypeListImpl<int>>>>>::value,
+              "");
+static_assert(std::is_same<TypeListImpl<int, double>, Flatten<TypeListImpl<int, double>>>::value, "");
+static_assert(std::is_same<TypeListImpl<int, double>, Flatten<TypeListImpl<TypeListImpl<int>, double>>>::value,
+              "");
+static_assert(std::is_same<TypeListImpl<int, double>,
+                           Flatten<TypeListImpl<TypeListImpl<TypeListImpl<int>>, double>>>::value,
+              "");
+
+// `DeduplicateImpl<TypeListImpl<LHS...>, TypeListImpl<REST...>>` appends the the types from `REST...`
```
Comment (mzhurovich) [inline Bricks/template/typelist.h]: s/the the/the/

File: Bricks/template/typelist.h [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+// that are not yet present in `LHS...` to `LHS...`.
+// It expects a flattened list of types as the second parameters.
```
Comment (mzhurovich) [inline Bricks/template/typelist.h]: s/parameters/parameter/?

File: Bricks/template/typelist.h [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+static_assert(!IsTypeList<double>::value, "");
+
+}  // namespace metaprogramming
+}  // namespace bricks
+
+// Export some symbols into global scope.
+using bricks::metaprogramming::TypeList;
+using bricks::metaprogramming::TypeListContains;
+using bricks::metaprogramming::IsTypeList;
+using bricks::metaprogramming::TypeListSize;
```
Comment (mzhurovich) [inline Bricks/template/typelist.h]: Do we now have a policy to inject methods from `*.h`-s into global scope? Or it's a special case?

Comment (dkorolev) [inline Bricks/template/typelist.h]: No policy yet :-(

I think eventually we'll do it for generic names, like we did for `HTTP`/`GET`/`POST`/`Request`/`Response`.

An idea I have in mind is to have a `current.h` header that `#include`-s major Current modules and does a few dozen `using`-s in the global namespace.

For now I'd be fine either way :-)

File: Bricks/template/typelist.h (hunk 1 above)
Comment (dkorolev) [inline Bricks/template/typelist.h]: :+1: 7159424

File: Bricks/template/typelist.h (hunk 2 above)
Comment (dkorolev) [inline Bricks/template/typelist.h]: :+1: 7159424

**Comment by dkorolev** [comment]
@mzhurovich Thank you, PTAL once you have a minute. Thanks!

File: Bricks/template/typelist.h (hunk 3 above)
Comment (mzhurovich) [inline Bricks/template/typelist.h]: Got it, thnx.

**Comment by dkorolev** [comment]
Thank you @mzhurovich !

**Pull request merged**